src/arraymancer/laser/tensor/initialization.nim

```diff
@@ -250,6 +257,18 @@ proc toTensor*[T](a: openArray[T]): auto =
   let data = toSeq(flatIter(a))
   result = toTensor(data, shape)
 
+proc toTensor*[T](a: SomeSet[T]): auto =
+  ## Convert a HashSet or an OrderedSet into a Tensor
+  ##
+  ## Input:
+  ##      - An HashSet or an OrderedSet
+  ## Result:
+  ##      - A Tensor of the same shape
+  var shape = MetaData()
+  shape.add(a.len)
+  let data = toSeq(a)
+  result = toTensor(data, shape)
```

Comment:
I suppose we could also do:
```nim
result = newTensorUninit[T](a.card)
for i, x in a:
  result[i] = x
```
to avoid the double copy (set -> seq -> tensor)

Reply:
Makes sense. I'll make that change

Reply:
Interestingly a performance measurement using `timeit` showed that the original version was faster on Windows. We will leave this as is for now while we investigate what is going on (and if we find out why we'll update this in a separate PR).

Reply:
Yeah, let's investigate it further after this PR.